integrationTests/multiShard/block/executingRewardMiniblocks_test.go

```diff
@@ -297,7 +303,37 @@ func updateNumberTransactionsProposed(
 	transactionsForLeader[addressProposer] += nbTransactions
 }
 
-func verifyRewards(
+func updateRewardsForMetachain(rewardsMap map[string]uint32, consensusNode *integrationTests.TestProcessorNode) {
+	metaRewardDataSlice := consensusNode.SpecialAddressHandler.ConsensusMetaRewardData()
+	if len(metaRewardDataSlice) > 0 {
+		for _, metaRewardData := range metaRewardDataSlice {
+			for _, addr := range metaRewardData.Addresses {
+				rewardsMap[addr]++
+			}
+		}
+	}
+}
+
+func verifyRewardsForMetachain(
+	t *testing.T,
+	mapRewardsForMeta map[string]uint32,
+	nodes map[uint32][]*integrationTests.TestProcessorNode,
+) {
+
+	// TODO this should be read from protocol config
+	rewardValue := uint32(1000)
+
+	for metaAddr, numOfTimesRewarded := range mapRewardsForMeta {
+		addrContainer, _ := integrationTests.TestAddressConverter.CreateAddressFromPublicKeyBytes([]byte(metaAddr))
+		acc, err := nodes[0][0].AccntState.GetExistingAccount(addrContainer)
+		assert.Nil(t, err)
+
+		expectedBalance := big.NewInt(int64(numOfTimesRewarded * rewardValue))
+		assert.Equal(t, expectedBalance, acc.(*state.Account).Balance)
```

Comment:
test should be done on value, not on pointer. Use big.Int compare function

Reply:
assert.Equal should be fine as it checks the value, otherwise also the test would fail because the expected balance is a new instance.

Reply:
agree. didn't change